### ISSUE TITLE
Support Flags for positional arguments and request bodies

### ISSUE THREAD

**Issue by thebauda** [danielgtaylor/restish#236]
**TLDR; Support accessing positional arguments and request bodies as flags**

When a restish command is available that was generated from an endpoint that path parameters and/or takes a request body, the command is structured as so:

`restish <api> <command> <positional-argument-1> '{input.json}'`

For example, the following endpoint with the following request body
`POST my-api/v2/orgs/{org_id}/create_user_in_organization `
```
{
  "first_name": string,
  "last_name": string
}
```
would be called by restish as so
```
restish my-api create-user-in-organization <org-id> '{"first_name": "foo","last_name": "bah"}'
```

It would be better if restish could support using flags to pass values to positional arguments and request bodies. The above example could then be called as so:
```
restish my-api create-user-in-organization --org-id <org-id> --first-name "foo" last-name "bah"
```

**Comment by danielgtaylor** [comment]
@thebauda why would this be better? Flags are typically optional while the `org-id` positional parameter is always required in your example. You can also use the [CLI shorthand](https://rest.sh/#/shorthand) for the params to save yourself having to type out JSON like this:

```sh
restish my-api create-user-in-organization MyOrg first_name: foo, last_name: bah
```

**Comment by thebauda** [comment]
> @thebauda why would this be better? Flags are typically optional while the `org-id` positional parameter is always required in your example. You can also use the [CLI shorthand](https://rest.sh/#/shorthand) for the body to save yourself having to type out JSON like this:
> 
> ```shell
> restish my-api create-user-in-organization MyOrg first_name: foo, last_name: bah
> ```

I'd like it where required and non-required items can be passed as flags. Not all of my users are familiar with CLI shorthand, and as you stated, writing JSON can be cumbersome and error-prone.

**Comment by thebauda** [comment]
Would you ever be willing to give my team and I a walkthrough on the Restish project architecture? I'd like to contribute PRs that introduce new features and have you review them.

**Comment by danielgtaylor** [comment]
Thanks for the request and the extra context. I’m going to close this as not planned for core/default generated command behavior for now.

I understand the motivation: flags can feel more familiar to users who do not know Restish shorthand and do not want to write JSON. The tradeoff is that generated commands can get very large and ambiguous pretty quickly. Required flags are also a slightly odd CLI fit, and body schemas bring a lot of edge cases: nested objects, arrays, `oneOf` / `anyOf`, multipart and non-JSON bodies, naming conflicts with Restish’s own `--rsh-*` flags, and help output that can become hard to scan on larger APIs.

For now I’d rather keep the core model focused on positional path params plus JSON/shorthand bodies. If more people ask for this, I’d be open to revisiting it as an opt-in generated operation mode, probably limited at first to path params and simple top-level JSON object fields, with clear fallbacks for complex request bodies rather than trying to flatten every schema into flags.
